### PR TITLE
PS cmdlets for application gateway L4 properties

### PR DESCRIPTION
**Pull request by gdhillon24**
## Description
PS cmdlets for application gateway L4 properties -> Added new comdlets for Listeners , BackendSettings, RoutingRules
Modified cmdlets.
New-AzApplicationGateway , Set-AzApplicationGatewayProbeConfig ,Add-AzApplicationGatewayProbeConfig ,New-AzApplicationGatewayProbeConfig 

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1119 }
        - {https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1211 }
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Network/Network/ApplicationGateway/BackendSettings/NewAzureApplicationGatewayBackendSettingsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApplicationGatewayBackendSetting"), OutputType(typeof(PSApplicationGatewayBackendSettings))]
```
Comment (VeryEarly) [inline src/Network/Network/ApplicationGateway/BackendSettings/NewAzureApplicationGatewayBackendSettingsCommand.cs:21]: this cmdlet will return ID like:
/subscriptions/{0}/resourceGroups/ResourceGroupNotSet/providers/Microsoft.Network/applicationGateways/ApplicationGatewayNameNotSet/{3}/{4}
can you please clarify in help message

**Comment by gdhillon24** [comment]
> It seems like backend setting related cmdlets only manipulate the in-memory PSApplicationGateway object without sending any request except Add-AzApplicationGatewayBackendSetting, can you please confirm why these cmdlets are needed.

Basically, the workflow to update BackendSettings would require to fetch the application gateway object and then use the new cmdlets to modify the backend settings in the application gateway object and at last update the Application Gateway Object using Set-AzApplicationGateway.

Same is true for Listeners and RoutingRules.

**Comment by gdhillon24** [comment]
> > It seems like backend setting related cmdlets only manipulate the in-memory PSApplicationGateway object without sending any request except Add-AzApplicationGatewayBackendSetting, can you please confirm why these cmdlets are needed.
> 
> Basically, the workflow to update BackendSettings would require to fetch the application gateway object and then use the new cmdlets to modify the backend settings in the application gateway object and at last update the Application Gateway Object using Set-AzApplicationGateway.
> 
> Same is true for Listeners and RoutingRules.

Example Commands:
$**appgw** = Get-AzApplicationGateway -Name $appgwName -ResourceGroupName $rgname
		$backendSettingGet = Get-AzApplicationGatewayBackendSetting -ApplicationGateway $**appgw** -Name $poolSetting01Name
		Add-AzApplicationGatewayBackendSetting -ApplicationGateway $appgw -Name $poolSetting02Name -Port 1234 -Protocol TCP  -Timeout 42
		Set-AzApplicationGatewayBackendSetting -ApplicationGateway $appgw -Name $poolSetting02Name -Port 123 -Protocol TCP  -Timeout 40
		Remove-AzApplicationGatewayBackendSetting -ApplicationGateway $appgw -Name $poolSetting02Name
		$appgw = Set-AzApplicationGateway -ApplicationGateway $appgw

And For **New-AzApplicationGatewayBackendSetting** it Outputs a **PSApplicationGatewayBackendSettings** which can be used to create a new ApplicationGateway.
Below is an Example.

		**$poolSetting01** = New-AzApplicationGatewayBackendSetting -Name $poolSetting01Name -Port 443 -Protocol TCP -Timeout 20
		$appgw = New-AzApplicationGateway -Name $appgwName -ResourceGroupName $rgname -Location $location -BackendAddressPools $pool -BackendSettingsCollection **$poolSetting01** -FrontendIpConfigurations $fipconfig -GatewayIpConfigurations $gipconfig -FrontendPorts $fp01 -Listeners $listener01 -RoutingRules $rule01 -Sku $sku -AutoscaleConfiguration $autoscaleConfig  -Probe $probe

File: src/Network/Network/ApplicationGateway/BackendSettings/NewAzureApplicationGatewayBackendSettingsCommand.cs (hunk 1 above)
Comment (gdhillon24) [reply src/Network/Network/ApplicationGateway/BackendSettings/NewAzureApplicationGatewayBackendSettingsCommand.cs:21]: **New-AzApplicationGatewayBackendSetting** it returns a **PSApplicationGatewayBackendSettings** Object
image

**Comment by VeryEarly** [comment]
> > > It seems like backend setting related cmdlets only manipulate the in-memory PSApplicationGateway object without sending any request except Add-AzApplicationGatewayBackendSetting, can you please confirm why these cmdlets are needed.
> > 
> > 
> > Basically, the workflow to update BackendSettings would require to fetch the application gateway object and then use the new cmdlets to modify the backend settings in the application gateway object and at last update the Application Gateway Object using Set-AzApplicationGateway.
> > Same is true for Listeners and RoutingRules.
> 
> Example Commands: **$appgw** = **Get-AzApplicationGateway** -Name $appgwName -ResourceGroupName $rgname
> 
> $backendSettingGet = **Get-AzApplicationGatewayBackendSetting** -ApplicationGateway **$appgw** -Name PoolSetting01Name
> 
> **Add-AzApplicationGatewayBackendSetting** -ApplicationGateway $appgw -Name $poolSetting02Name -Port 1234 -Protocol TCP -Timeout 42
> 
> **Set-AzApplicationGatewayBackendSetting** -ApplicationGateway $appgw -Name $poolSetting02Name -Port 123 -Protocol TCP -Timeout 40
> 
> **Remove-AzApplicationGatewayBackendSetting** -ApplicationGateway $appgw -Name $poolSetting02Name
> 
> $appgw = **Set-AzApplicationGateway** -ApplicationGateway $appgw
> 
> And For **New-AzApplicationGatewayBackendSetting** it Outputs a **PSApplicationGatewayBackendSettings** which can be used to create a new ApplicationGateway.
> 
> Below is an Example.
> 
> **$poolSetting01** = **New-AzApplicationGatewayBackendSetting** -Name $poolSetting01Name -Port 443 -Protocol TCP -Timeout 20
> 
> $appgw = **New-AzApplicationGateway** -Name $appgwName -ResourceGroupName $rgname -Location $location -BackendAddressPools $pool -BackendSettingsCollection **$poolSetting01** -FrontendIpConfigurations $fipconfig -GatewayIpConfigurations $gipconfig -FrontendPorts $fp01 -Listeners $listener01 -RoutingRules $rule01 -Sku $sku -AutoscaleConfiguration $autoscaleConfig -Probe $probe

make sense

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**